content/providers/05-observability/humanloop.mdx

```diff
@@ -0,0 +1,265 @@
+
+## Setup
+
+### Prerequisites
+
+- A Humanloop account and API key.
+  - [Sign up](https://app.humanloop.com/signup) or [login](https://app.humanloop.com/login) to Humanloop.
+  - Create an API key in [Organization Settings](https://app.humanloop.com/account/api-keys).
+- A Vercel AI SDK application.
+
+### Telemetry Configuration
+
+When sending traces to Humanloop, these parameters are added to the telemetry object:
+
+```ts
+experimental_telemetry: {
+  isEnabled: true,
+  functionId: 'unique-function-id', // Optional identifier for the function
+  metadata: {
+    humanloopPromptPath: 'Path/To/Prompt',
+    humanloopFlowPath: 'Path/To/Flow', // Optional
+    humanloopFlowId: 'flow-log-id'     // Optional
+  },
+}
+```
+
+### Environment Variables
+
+When using OpenTelemetry with Humanloop, the following environment variables configure the OTLP exporter:
+
+```bash
+OTEL_EXPORTER_OTLP_ENDPOINT=https://api.humanloop.com/v5/import/otel
+OTEL_EXPORTER_OTLP_PROTOCOL=http/json
+OTEL_EXPORTER_OTLP_HEADERS="X-API-KEY=xxxxxx" # Humanloop API key
+```
+
+## Framework Implementation
+
+<Tabs items={['Next.js', 'Node.js']}>
+  <Tab>
+    Next.js has support for OpenTelemetry instrumentation on the framework level. Learn more about it in the [Next.js OpenTelemetry guide](https://nextjs.org/docs/app/building-your-application/optimizing/open-telemetry).
+
+    Required dependencies:
+
+    <Tabs items={['pnpm', 'npm', 'yarn']}>
+      <Tab>
+        <Snippet
+          text="pnpm add @vercel/otel @opentelemetry/sdk-logs @opentelemetry/api-logs @opentelemetry/instrumentation"
+          dark
+        />
+      </Tab>
+      <Tab>
+        <Snippet
+          text="npm install @vercel/otel @opentelemetry/sdk-logs @opentelemetry/api-logs @opentelemetry/instrumentation"
+          dark
+        />
+      </Tab>
+      <Tab>
+        <Snippet
+          text="yarn add @vercel/otel @opentelemetry/sdk-logs @opentelemetry/api-logs @opentelemetry/instrumentation"
+          dark
+        />
+      </Tab>
+    </Tabs>
+
+    Update your `.env.local` file to configure the OTLP Exporter:
+
+    ```bash filename=".env.local"
+    OTEL_EXPORTER_OTLP_ENDPOINT=https://api.humanloop.com/v5/import/otel
+    OTEL_EXPORTER_OTLP_PROTOCOL=http/json
+    OTEL_EXPORTER_OTLP_HEADERS="X-API-KEY=xxxxxx" # Humanloop API key
+    ```
+
+    Register the OpenTelemetry SDK `instrumentation.ts` file (in root or src/ dir):
+
+    ```ts filename="instrumentation.ts"
+    import { registerOTel } from '@vercel/otel';
+
+    export function register() {
+      registerOTel({
+        serviceName: 'humanloop-vercel-ai-nextjs',
+      });
+    }
+    ```
+
+    Your calls to the AI SDK should now be logged to Humanloop.
+
+  </Tab>
+  <Tab>
+
+    ### Node.js Implementation
+
+    OpenTelemetry has a package to auto-instrument Node.js applications. Learn more about it in the [OpenTelemetry Node.js guide](https://opentelemetry.io/docs/languages/js/getting-started/nodejs/).
```

Comment:
"Add OpenTelemetry to your Node.js project. To learn more it, check out the [....."